src/dependency-manager/src/utils/validators/dictionary_types.ts

```diff
@@ -0,0 +1,29 @@
+import { registerDecorator, ValidationArguments, ValidationOptions } from "class-validator";
+
+/** Require each value of the object is of one of the types specified */
+export const DictionaryTypes = (types: string[], validationOptions?: ValidationOptions) => {
```

Comment:
We probably don't need to support a list of types.
I like the solution though. Better to be explicit.